runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java

```diff
@@ -117,21 +133,100 @@ public void onScriptEvent(ScriptCallbackEvent event)
 		}
 	}
 
+	@Subscribe
+	public void onVarbitChanged(VarbitChanged event)
+	{
+		if (client.getWidget(WidgetInfo.SETTINGS_TAB_SCROLL_WHEEL_ICON) == null || !zoomConfig.controlZoom())
+		{
+			return;
+		}
+
+		loggedIn = true;
+
+		if (!hotkeyDown)
+		{
+			updateControlZoom(false);
+		}
+		else
+		{
+			updateControlZoom(true);
+		}
+	}
+
+	@Override
+	public void keyPressed(KeyEvent e)
+	{
+		if (!zoomConfig.controlZoom())
+		{
+			return;
+		}
+
+		if (e.getKeyCode() == KeyEvent.VK_CONTROL)
+		{
+			hotkeyDown = true;
+			updateControlZoom(true);
+		}
+	}
+
+	@Override
+	public void keyReleased(KeyEvent e)
+	{
+		if (!zoomConfig.controlZoom())
+		{
+			return;
+		}
+
+		if (e.getKeyCode() == KeyEvent.VK_CONTROL)
+		{
+			hotkeyDown = false;
+			updateControlZoom(false);
+		}
+	}
+
+	@Override
+	public void keyTyped(KeyEvent e)
+	{
+
+	}
+
 	@Override
 	protected void startUp()
 	{
 		client.setCameraPitchRelaxerEnabled(zoomConfig.relaxCameraPitch());
+		keyManager.registerKeyListener(this);
+		updateControlZoom(!zoomConfig.controlZoom());
 	}
 
 	@Override
 	protected void shutDown()
 	{
 		client.setCameraPitchRelaxerEnabled(false);
+		keyManager.unregisterKeyListener(this);
+		updateControlZoom(true);
+	}
+
+	private void updateControlZoom(Boolean value)
+	{
+		if (!loggedIn)
+		{
+			return;
+		}
+
+		client.getWidget(WidgetInfo.SETTINGS_TAB_SCROLL_WHEEL_INDICATOR).setHidden(value);
+		client.setVarbitValue(client.getVarps(), Varbits.SCROLL_WHEEL_ZOOM.getId(), value ? 0 : 1);
```

Comment:
this is supposed to be only for devtools

Reply:
This was the simplest way I could think off while still allowing scroll to be used for other things than banking, and keep the "zoom bar" working